lib/src/selection.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Where a selection start or end point may appear in some piece of text.
+abstract class Selection {
+  /// The chunk of text.
+  String get text;
+
+  /// The offset from the beginning of [text] where the selection starts, or
+  /// `null` if the selection does not start within this chunk.
+  int? get selectionStart => _selectionStart;
```

Comment:
This file has the word "selection" a lot...

Can we avoid repeating the class name context in it's members?

```dart
int? get start;
int? get end;
void startAt(int start);
void startBackFromTextEnd(int fromEnd);
void endAt(int end);
void endBackFromTextEnd(int fromEnd);
```

Reply:
It looks a little weird here, but this class is used as a base class for `Chunk` and `SourceComment` and in the context of those, it makes sense for the members to have more explicit selection-related names.